library/alloc/src/vec.rs

```diff
@@ -194,6 +194,22 @@ use crate::raw_vec::RawVec;
 /// can be slow. For this reason, it is recommended to use [`Vec::with_capacity`]
 /// whenever possible to specify how big the vector is expected to get.
 ///
+/// In summary, a vector containing 1, 2 with capacity 4 can be visualized as:
+/// ```text
+/// Stack      ptr   capacity    len
+///        +--------+--------+--------+
+///        | 0x0123 |      4 |      2 |
+///        +--------+--------+--------+
```

Comment:
```suggestion
///        ┌────────┬────────┬────────┐
///        │ 0x0123 │      4 │      2 │
///        └────────┴────────┴────────┘
```

How about Unicode box drawing characters? I think they are more appropriate and prettier. 
(Same applies for the box below, of course)

Reply:
I don't choose unicode because I think they are only prettier on some fonts. In the terminal, not everyone uses a terminal that draw boxes correctly, there might be some gaps between the boxes, and even on the browser that looks a bit ugly when it wasn't rendered the way it is supposed to do. At the very least, using ASCII makes it looks consistent across devices and screens, so everyone get not perfect pixels but still good enough for eyes to read.

Browser does not render correctly for me

![image](https://user-images.githubusercontent.com/4687791/96148574-f6ea4480-0f3a-11eb-916f-3094d0c3e268.png)

If you insist, I could add it but I don't think they are improvement all the times, maybe only on some devices.

Reply:
Mhhh fair points. I still prefer the proper box drawing characters, but that's not a strong opinion. We can keep the ASCII ones in this PR.